### PR TITLE
Allow customize locales resolver

### PR DESCRIPTION
**Pull request by ratson**
It should be easy to implement switch language feature without re-implementing all loading logic.

### CODE REVIEW

**Comment by KevinGrandon** [comment]
Thanks for the pull request, I'll get this on our list of items to review.

**Comment by ganemone** [comment]
Looks like there are some linting errors

File: src/loader.js [hunk 1]
```diff
@@ -21,7 +21,9 @@ export type I18nLoaderType = {
   ) => {locale: string | Locale, translations: TranslationsObjectType},
 };
 
-const loader: () => I18nLoaderType = () => {
+const defaultResolveLocales = (ctx: Context) => ctx.headers['accept-language'];
+
+const loader: () => I18nLoaderType = (resolveLocales = defaultResolveLocales) => {
```
Comment (AlexMSmithCA) [inline src/loader.js]: Can we update the explicit type definition for `loader` here?  It is currently `() => I18nLoaderType` but now expects an optional `resolveLocales` argument.

As a starting point, maybe something like:
```
type LoaderFactoryType = (resolver?: (ctx: Context) => string) => I18nLoaderType;
```

**Comment by AlexMSmithCA** [comment]
Bump.  @ratson could you take a look at the failures here?

**Comment by chrisdothtml** [comment]
I've gone ahead and fixed the lint/typing issues, as well as added a test and some docs. This should be ready for re-review 👍

**Pull request merged**